### PR TITLE
virtctl: Deprecate duplicated volume import flags

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

This PR deprecates the `--volume-clone-pvc`, `--volume-datasource` and `--volume-blank` flags in favor of the `--volume-import` flag, which provides the same functionality. The flags are still kept and aliases to the new flag.

The `--volume-import` flag is updated to match the functionality that was found previously in the other flags:

- The type `ds` was added to the `--volume-import` flag
- Support for param `bootorder` was added to the `--volume-import` flag
- Error messages were adjusted for consistency

Before this PR:

virtctl create vm has multiple flags to import volumes with varying types.

After this PR:

virtctl create vm has a single flag to import volumes. This improves the UX by making it more clear.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
virtctl: The flags `--volume-clone-pvc`, `--volume-datasource` and `--volume-blank` are deprecated in favor of the `--volume-import` flag.
```

### CODE REVIEW

**Comment by lyarwood** [comment]
Thanks @0xFelix  but I think it would be better to deprecate for a release and then remove with the alternatives clearly documented during both steps in our help output somehow.

**Comment by 0xFelix** [comment]
I'll see what I can do. Perhaps it is possible to make the old flags shortcuts to the new flag and to print a deprecation notice.

**Review by lyarwood — COMMENTED** [review]
This lgtm at first glance but I'm confused by one of the early commits. b256b496ac2d4e7c455370fc437a9b5cea033971 changes the ordering of volumes being added to the VM but doesn't say why it's changed?

**Comment by 0xFelix** [comment]
Right, it is because aliasing the deprecated commands to volume import will change their order. Once the flags are dropped only 'Imported volumes' will remain and Containerdisks and PVC will always come first in the order.

**Comment by 0xFelix** [comment]
Updated the commit description

**Review by lyarwood — APPROVED** [review]
/lgtm

Thanks for fixing up that commit message, looks good otherwise!

**Comment by alicefr** [comment]
Looks good
/approve

File: pkg/virtctl/create/vm/vm.go [hunk 1]
```diff
@@ -190,27 +199,30 @@ func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
...
 	))
 
 	cmd.Flags().StringVar(&c.cloudInitUserData, CloudInitUserDataFlag, c.cloudInitUserData, "Specify the base64 encoded cloud-init user data of the VM.")
 	cmd.Flags().StringVar(&c.cloudInitNetworkData, CloudInitNetworkDataFlag, c.cloudInitNetworkData, "Specify the base64 encoded cloud-init network data of the VM.")
 
+	// Deprecated flags
+	cmd.Flags().StringArrayVar(&c.dataSourceVolumes, DataSourceVolumeFlag, c.dataSourceVolumes, "Specify a DataSource to be cloned by the VM. Can be provided multiple times.\nSupported parameters: name:string,src:string,bootorder:uint,size:resource.Quantity\nDEPRECATED: Use --volume-import instead")
+	cmd.Flags().StringArrayVar(&c.clonePvcVolumes, ClonePvcVolumeFlag, c.clonePvcVolumes, "Specify a PVC to be cloned by the VM. Can be provided multiple times.\nSupported parameters: name:string,src:string,bootorder:uint,size:resource.Quantity\nDEPRECATED: Use --volume-import instead")
+	cmd.Flags().StringArrayVar(&c.blankVolumes, BlankVolumeFlag, c.blankVolumes, "Specify a blank volume to be used by the VM. Can be provided multiple times.\nSupported parameters: name:string,size:resource.Quantity\nDEPRECATED: Use --volume-import instead")
```
Comment (jcanocan) [inline pkg/virtctl/create/vm/vm.go:224]: maybe in the hint for the user, you can add the complete command to do the same, e.g. "`volume-datasource` -> Use `volume-import=type:ds` instead."

**Review by jcanocan — COMMENTED** [review]
Nice job!
Just a small suggestion.

**Comment by 0xFelix** [comment]
@jcanocan Had to rebase this, can you PTAL again?

**Pull request merged**